### PR TITLE
Add wolfi image

### PR DESCRIPTION
**Pull request by joonas**
## Feature or Problem

This adds [wolfi](https://github.com/wolfi-dev/) as an alternative base image for the wasmCloud host and wash OCI images.

### CODE REVIEW

**Review by brooksmtownsend — COMMENTED** [review]
All looks solid to me at a glance, one question but I will defer to @rvolosatovs for an extra look

File: flake.nix [hunk 1]
```diff
@@ -277,6 +277,30 @@
...
+
+          wolfi.aarch64 = pullWolfi {
+            imageDigest = "sha256:e8c9aae5f4f1cddf9ed0449f626772a55a1a45e86155698c9d6e34c862b79736";
+            sha256 = "0wjpvlnashghp5d9vxj9s6685rvahxvd7ygndsjmq3nr9h1xbw2x";
+          };
+          wolfi.x86_64 = pullWolfi {
+            imageDigest = "sha256:4e00b653d9ad1cc7c82856d827ec003e496747d608fd842196fc888f39ddc59d";
+            sha256 = "1rbm5afznmbhrx8xhkx7pbxkjmw1aqxfh3wwm7sdf9w9n1qbjbw7";
+          };
+
```
[the remark this answers is not in the document]
Comment (joonas) [reply flake.nix:303]: Great question!

From what I've seen, the latest tag digests are kept around for a while based on a cursory look at the history of `cgr.dev/chainguard/wolfi-base:latest` with some help from crane:

```shell
# List the current digest pointing to `latest` tag:
$ crane digest cgr.dev/chainguard/wolfi-base:latest
sha256:35c767080978768b86904e6c64845736c3cf05c406299b944dcd24ffe8270df5

# Iterate through 10 randomly picked sha256 digests, check if they still exist and log if they do not:
$ for i in $(seq 10); do
    RANDOM_DIGEST=$(crane ls cgr.dev/chainguard/wolfi-base | grep sha256- | shuf -n 1 | awk -F "[-.]" '{print $2}' | tr -d '\n')
    docker manifest inspect "cgr.dev/chainguard/wolfi-base@sha256:${RANDOM_DIGEST}" 2>/dev/null 1>&2
    code=$?
    if [ $code -eq 1 ]; then
        echo "Manifest not found for cgr.dev/chainguard/wolfi-base@sha256:${RANDOM_DIGEST}"
        break
    fi
done

# And to look at one example of how long these digests have been around:
$ crane config "cgr.dev/chainguard/wolfi-base@sha256:${RANDOM_DIGEST}" | jq .
{
  "architecture": "amd64",
  "author": "github.com/chainguard-dev/apko",
  "created": "2023-03-31T22:50:06Z",
  "history": [
    {
      "author": "apko",
      "created": "2023-03-31T22:50:06Z",
      "created_by": "apko",
      "comment": "This is an apko single-layer image"
    }
  ],
  "os": "linux",
  "rootfs": {
    "type": "layers",
    "diff_ids": [
      "sha256:150554a87dc967a222cfff9054b787caec9855cb76751cc32602c86985b22eaa"
    ]
  },
  "config": {
    "Cmd": [
      "/bin/sh",
      "-l"
    ],
    "Env": [
      "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
      "SSL_CERT_FILE=/etc/ssl/certs/ca-certificates.crt"
    ]
  }
}
```

So the long and short of it is that:
1. Given long enough period of time, that tag _could_ become invalid, but on a cursory look, that could/should be over 12 months - we should probably update more often than that.
2. If we're worried about that happening, we should reach out to the Chainguard folks and talk about what they would recommend doing here.

[the remark this answers is not in the document]
Comment (rvolosatovs) [reply flake.nix:303]: IIRC, these will never become invalid as long as the actual artifacts are still accessible (or cached) - this functionality uses `skopeo` under the hood, which fetches by digest. In other words, the tag value should never be actually used for fetching

**Review by rvolosatovs — APPROVED** [review]
Nice work!

File: flake.nix (hunk 1 above)
[the remark this answers is not in the document]
Comment (joonas) [reply flake.nix:303]: Thanks for the clarification and added context @rvolosatovs!

**Review by brooksmtownsend — APPROVED** [review]

**Pull request merged**